File: Dumplings/src/main/java/net/sinclairstudios/android/dumplings/domain/Dumpling.java
package net.sinclairstudios.android.dumplings.domain;

import java.io.Serializable;

public class Dumpling implements Serializable {
    private final String name;

    public Dumpling(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return name + " dumpling";
    }
}

File: Dumplings/src/main/kotlin/net/sinclairstudios/android/dumplings/domain/Dumpling.kt
package net.sinclairstudios.android.dumplings.domain
import java.io.Serializable

public open class Dumpling(val name: String): Serializable {

    public override fun toString(): String {
        return "$name dumpling"
    }

    public override fun equals(that: Any?): Boolean {
        return that is Dumpling
            && equals(that)
    }

    public fun equals(that: Dumpling): Boolean {
        return this.name.equals(that.name)
    }
}
